doc/builders/images/portableservice.section.md

```diff
@@ -0,0 +1,64 @@
+
+`pkgs.portableService` is a function to create **"portable service images"**,
+as read-only, immutable, `squashfs` archives.
+
+systemd supports a concept of ["Portable Services"](https://systemd.io/PORTABLE_SERVICES/).
+"Portable Services" are a delivery method for system services that uses two specific features of container management:
+
+* Applications are bundled. I.e. multiple services, their binaries and
+  all their dependencies are packaged in an image, and are run directly from it.
+* Stricter default security policies, i.e. sandboxing of applications.
+
+This allows to use the benefits of nix, to build images for use on many recent Linux distributions.
+
+The primary tool for interacting with Portable Services is `portablectl`,
+and they are managed by the `systemd-portabled` system service.
+
+> **_NOTE:_** Portable services were introduced and only supported since systemd 239 (released on 2018-06-22).
+
+A very simple example of using `portableService` is described below:
+
+[]{#ex-pkgs-portableService}
+
+```nix
+pkgs.portableService {
+  pname = "demo";
+  version = "1.0";
+  units = [ demo-service demo-socket ];
+}
+```
+
+The above example will build an squashfs archive image in `result/$pname_$version.raw`. The image will contain the
+file-system structure as required by the portable service specification, and a sub-set of the nix store with all the
+dependencies of the two derivations in the `units` list.
+`units` must be a list of derivations, and their names must be prefixed with the service name ("demo" in this case),
+otherwise `systemd-portabled` will ignore them.
+
+> **_NOTE:_** The `.raw` file extension of the image is required by the portable services specification.
+
+Some other options available are:
+- `description`, `homepage`: are added to the `/etc/os-release` in the image and are shown by the portable services tooling.
+   Default to empty values, not added to os-release.
+- `symlinks`:  a list of attribute sets {object, symlink}. Symlinks will be created  in the root filesystem of the image to
+   objects in the nix store. Defaults to an empty list.
+- `contents`: a list of additional derivations to be included in the image nix store, as-is. Defaults to an empty list.
+- `squashfsTools`: defaults to `pkgs.squashfsTools`, allows you to override the package that provides `mksquashfs`.
+- `squash-compression`, `squash-block-size`: options to `mksquashfs`. Default to `"xz -Xdict-size 100%"` and `"1M"` respectively.
+
+A typical usage of `symlinks` would be:
+```nix
+  symlinks = [
+    { object = "${pkgs.cacert}/etc/ssl"; symlink = "/etc/ssl"; }
+    { object = "${pkgs.bash}/bin/bash"; symlink = "/bin/sh"; }
+    { object = "${pkgs.php}/bin/php"; symlink = "/usr/bin/php"; }
+  ];
+```
+to create these symlinks for legacy applications that assume them existing globally.
+
+Once the image is created, and deployed on a host in `/var/lib/portables/`, you can use it as:
+```sh
+# portablectl attach demo_1.0.raw
```

Comment:
Does that have to be run as root? It will render as a comment right now. I suggest not adding prompts and describing in the text that root is required. Then the console input can be pasted verbatim.

Reply:
> Does that have to be run as root? 

yes

> I suggest not adding prompts and describing in the text that root is required. Then the console input can be pasted verbatim.

👍